duckduckgo/macos-browser
# Refactor Overlay Pixels

Task/Issue URL: https://app.asana.com/0/1204099484721401/1208674332470432/f
Tech Design URL:
CC:

**Description**:
Refactors Overlay Pixels to use WebKit KVO and backForward lists for improved reliability


**Steps to test this PR**:
1. Filter the console messages for "PixelKit."
2. Go to Youtube.com
3. Search for MKBHD
4. Select a video
5. Hit Back
6. Confirm  `m_mac_duckplayer_youtube_overlay_navigation_back` pixel fires
7. Select another video
8. Hit Refresh
9. Confirm `m_mac_duckplayer_youtube_overlay_navigation_refresh`pixel fires
10. Navigate to other Youtube video suggestion
11. Confirm `m_mac_duckplayer_youtube_overlay_navigation_within-youtube`
12. Visit another website (i.e. example.com)
13. Confirm `m_mac_duckplayer_youtube_overlay_navigation_outside-youtube` fires

Smoke test going to/from videos and confirm pixels fire as expected.  

Demo:
![12-06 at 19 35](https://github.com/user-attachments/assets/5c5b1c45-2c8f-48d9-802c-f2fd90e6fc0c)
